src/to-string.cpp

```diff
@@ -498,60 +506,61 @@ const char * get_string( rs2_camera_info value )
 #undef CASE
 }
 
-const char * get_string( rs2_frame_metadata_value value )
+std::string const & get_string( rs2_frame_metadata_value value )
 {
-#define CASE( X ) STRCASE( FRAME_METADATA, X )
-    switch( value )
+    static auto str_array = []()
     {
-    CASE( FRAME_COUNTER )
-    CASE( FRAME_TIMESTAMP )
-    CASE( SENSOR_TIMESTAMP )
-    CASE( ACTUAL_EXPOSURE )
-    CASE( GAIN_LEVEL )
-    CASE( AUTO_EXPOSURE )
-    CASE( WHITE_BALANCE )
-    CASE( TIME_OF_ARRIVAL )
-    CASE( TEMPERATURE )
-    CASE( BACKEND_TIMESTAMP )
-    CASE( ACTUAL_FPS )
-    CASE( FRAME_LASER_POWER )
-    CASE( FRAME_LASER_POWER_MODE )
-    CASE( EXPOSURE_PRIORITY )
-    CASE( EXPOSURE_ROI_LEFT )
-    CASE( EXPOSURE_ROI_RIGHT )
-    CASE( EXPOSURE_ROI_TOP )
-    CASE( EXPOSURE_ROI_BOTTOM )
-    CASE( BRIGHTNESS )
-    CASE( CONTRAST )
-    CASE( SATURATION )
-    CASE( SHARPNESS )
-    CASE( AUTO_WHITE_BALANCE_TEMPERATURE )
-    CASE( BACKLIGHT_COMPENSATION )
-    CASE( GAMMA )
-    CASE( HUE )
-    CASE( MANUAL_WHITE_BALANCE )
-    CASE( POWER_LINE_FREQUENCY )
-    CASE( LOW_LIGHT_COMPENSATION )
-    CASE( FRAME_EMITTER_MODE )
-    CASE( FRAME_LED_POWER )
-    CASE( RAW_FRAME_SIZE )
-    CASE( GPIO_INPUT_DATA )
-    CASE( SEQUENCE_NAME )
-    CASE( SEQUENCE_ID )
-    CASE( SEQUENCE_SIZE )
-    CASE( TRIGGER )
-    CASE( PRESET )
-    CASE( INPUT_WIDTH )
-    CASE( INPUT_HEIGHT )
-    CASE( SUB_PRESET_INFO )
-    CASE( CALIB_INFO )
-    CASE( CRC )
-
-    default:
-        assert( ! is_valid( value ) );
-        return UNKNOWN_VALUE;
-    }
+        std::vector< std::string > arr( RS2_FRAME_METADATA_COUNT );
+#define CASE( X ) STRARR( arr, FRAME_METADATA, X );
+        CASE( FRAME_COUNTER )
```

Comment:
Its a matter of style but I prefer not to define and undefine CASE, and probably doing so for all other types as well.
You can change `CASE( FRAME_COUNTER )` to `STRARR( arr, FRAME_METADATA, FRAME_COUNTER )` and you can do it easily for all the cases here at once in VS using vertical editing.

Reply:
Replied on call :)